### PR TITLE
Added unit tests for GANs

### LINKED ISSUE #815: Add unit testing for GANs
### Is your feature request related to a problem? Please describe.
Pretty self explanatory - we need to test the GAN utilities
### Describe the solution you'd like
Incorporation of the GAN unit tests into main testing file, incorporating testing data that will be automatically downloaded (this data is basically the same for all modalities as we have it now, but with mask files removed).
### Describe alternatives you've considered
Merge it into the main GANDLF tests in the future.

### PR DESCRIPTION
**Pull request by szmazurek**
Fixes #815

## Proposed Changes
- added tests for all modalities and dimensions
- data downloading and construction needs to probably get updated, as sytnehis data requires to be downloaded

## Checklist
- [X] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [X] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [X] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [X] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [X] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [ ] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [X] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].

### CODE REVIEW

**Comment by sarthakpati** [comment]
I will wait for the tests to pass before reviewing this. 😄

**Comment by szmazurek** [comment]
> I will wait for the tests to pass before reviewing this. 😄

I think that they will fail anyway due to fact that there are two test_full files (one in GAN). Do you think I should unify already? If so, the download function would need to also download new unit test data (or copy existing one, because it is the same virtually). The error is due to the fact that pytest just cannot start.
EDIT: I made a commit that fixes that - GAN unit tests are not yet being executed, as we do not have full functionalities merged yet. They are temporairly disabled then

File: GANDLF/GAN/testing/noexectest_gan.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+            with zipfile.ZipFile(output, "r") as zip_ref:
+                zip_ref.extractall(testingDir)
+            os.remove(output)
+            break
+    sanitize_outputDir()
+
+    print("passed")
+
+
+def test_generic_constructTrainingCSV():
```
Comment (sarthakpati) [inline GANDLF/GAN/testing/noexectest_gan.py:63]: We need to think about consolidating functionality from here and the current tests.

**Review by sarthakpati — APPROVED** [review]
Sounds good, thanks.

File: GANDLF/GAN/testing/noexectest_gan.py (hunk 1 above)
Comment (szmazurek) [reply GANDLF/GAN/testing/noexectest_gan.py:63]: yes, definetely -I believe integrating tests will be of the top priority when all pipeline is finally merged

Comment (sarthakpati) [reply GANDLF/GAN/testing/noexectest_gan.py:63]: Cool, thanks.

**Pull request merged**